### ISSUE TITLE
HF Callback not working with Urdu Text

### ISSUE THREAD

**Issue by kingabzpro** [gradio-app/gradio#841]
### Describe the bug

The call back works fine with vision and chatbot but with ASR it's generating error and I cant seem to figure it out. 

### Reproduction

Link to app: https://huggingface.co/spaces/kingabzpro/Urdu-ASR-SOTA
Link to dataset: https://huggingface.co/datasets/kingabzpro/Urdu-ASR-flags/tree/main

### Screenshot

![Uploading image.png…]()

### Logs

```shell
ERROR:    Exception in ASGI application
Traceback (most recent call last):
  File "/home/user/.local/lib/python3.8/site-packages/uvicorn/protocols/http/h11_impl.py", line 366, in run_asgi
    result = await app(self.scope, self.receive, self.send)
  File "/home/user/.local/lib/python3.8/site-packages/uvicorn/middleware/proxy_headers.py", line 75, in __call__
    return await self.app(scope, receive, send)
  File "/home/user/.local/lib/python3.8/site-packages/fastapi/applications.py", line 261, in __call__
    await super().__call__(scope, receive, send)
  File "/home/user/.local/lib/python3.8/site-packages/starlette/applications.py", line 112, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/home/user/.local/lib/python3.8/site-packages/starlette/middleware/errors.py", line 181, in __call__
    raise exc
  File "/home/user/.local/lib/python3.8/site-packages/starlette/middleware/errors.py", line 159, in __call__
    await self.app(scope, receive, _send)
  File "/home/user/.local/lib/python3.8/site-packages/starlette/middleware/cors.py", line 92, in __call__
    await self.simple_response(scope, receive, send, request_headers=headers)
  File "/home/user/.local/lib/python3.8/site-packages/starlette/middleware/cors.py", line 147, in simple_response
    await self.app(scope, receive, send)
  File "/home/user/.local/lib/python3.8/site-packages/starlette/exceptions.py", line 82, in __call__
    raise exc
  File "/home/user/.local/lib/python3.8/site-packages/starlette/exceptions.py", line 71, in __call__
    await self.app(scope, receive, sender)
  File "/home/user/.local/lib/python3.8/site-packages/fastapi/middleware/asyncexitstack.py", line 21, in __call__
    raise e
  File "/home/user/.local/lib/python3.8/site-packages/fastapi/middleware/asyncexitstack.py", line 18, in __call__
    await self.app(scope, receive, send)
  File "/home/user/.local/lib/python3.8/site-packages/starlette/routing.py", line 656, in __call__
    await route.handle(scope, receive, send)
  File "/home/user/.local/lib/python3.8/site-packages/starlette/routing.py", line 259, in handle
    await self.app(scope, receive, send)
  File "/home/user/.local/lib/python3.8/site-packages/starlette/routing.py", line 61, in app
    response = await func(request)
  File "/home/user/.local/lib/python3.8/site-packages/fastapi/routing.py", line 227, in app
    raw_response = await run_endpoint_function(
  File "/home/user/.local/lib/python3.8/site-packages/fastapi/routing.py", line 160, in run_endpoint_function
    return await dependant.call(**values)
  File "/home/user/.local/lib/python3.8/site-packages/gradio/routes.py", line 283, in flag
    await run_in_threadpool(
  File "/home/user/.local/lib/python3.8/site-packages/starlette/concurrency.py", line 39, in run_in_threadpool
    return await anyio.to_thread.run_sync(func, *args)
  File "/home/user/.local/lib/python3.8/site-packages/anyio/to_thread.py", line 28, in run_sync
    return await get_asynclib().run_sync_in_worker_thread(func, *args, cancellable=cancellable,
  File "/home/user/.local/lib/python3.8/site-packages/anyio/_backends/_asyncio.py", line 818, in run_sync_in_worker_thread
    return await future
  File "/home/user/.local/lib/python3.8/site-packages/anyio/_backends/_asyncio.py", line 754, in run
    result = context.run(func, *args)
  File "/home/user/.local/lib/python3.8/site-packages/gradio/flagging.py", line 381, in flag
    filepath = component.save_flagged(
  File "/home/user/.local/lib/python3.8/site-packages/gradio/inputs.py", line 1295, in save_flagged
    return self.save_flagged_file(
  File "/home/user/.local/lib/python3.8/site-packages/gradio/component.py", line 60, in save_flagged_file
    file = processing_utils.decode_base64_to_file(data, encryption_key)
  File "/home/user/.local/lib/python3.8/site-packages/gradio/processing_utils.py", line 187, in decode_base64_to_file
    data, extension = decode_base64_to_binary(encoding)
  File "/home/user/.local/lib/python3.8/site-packages/gradio/processing_utils.py", line 182, in decode_base64_to_binary
    data = encoding.split(",")[1]
IndexError: list index out of range
```

### System Info

```shell
HF Spaces, Edge browser
```

### Severity

annoyance

**Comment by aliabid94** [comment]
Taking a look.

**Comment by aliabid94** [comment]
@abidlabs actually can you take a look? It has to do with the HF callback

**Comment by abidlabs** [comment]
Yup I’ll take a look

**Comment by abidlabs** [comment]
Hi @kingabzpro, I tried to reproduce this issue, but was unable to, Specifically, can you try the following code locally and tell me if you run into any errors? (It's just a minimal version of your code)

```python
import gradio as gr

def asr(audio):
    return "hello"

HF_TOKEN = ""  # replace with your token
hf_writer = gr.HuggingFaceDatasetSaver(HF_TOKEN, "Urdu-ASR-flags2")


################### Gradio Web APP ################################

title = "Urdu Automatic Speech Recognition"

description = """
<p>
<center>
This model is a fine-tuned version of facebook/wav2vec2-xls-r-300m on the common_voice dataset.
</center>
</p>
<center>
<img src="https://huggingface.co/spaces/kingabzpro/Urdu-ASR-SOTA/resolve/main/Images/cover.jpg" alt="logo" width="550"/>
</center>
"""

article = "<p style='text-align: center'><a href='https://dagshub.com/kingabzpro/Urdu-ASR-SOTA' target='_blank'>Source Code on DagsHub</a></p><p style='text-align: center'><a href='https://huggingface.co/blog/fine-tune-xlsr-wav2vec2' target='_blank'>Fine-tuning XLS-R for Multi-Lingual ASR with 🤗 Transformers</a></p></center></p>"

examples = [["Sample/sample1.mp3"], ["Sample/sample2.mp3"], ["Sample/sample3.mp3"]]


Input = gr.inputs.Audio(
    source="microphone",
    type="filepath",
    optional=True,
    label="Please Record Your Voice",
)
Output = gr.outputs.Textbox(label="Urdu Script")

iface = gr.Interface(
    asr,
    Input,
    Output,
    title=title,
    flagging_options=["incorrect", "worst", "ambiguous"],
    flagging_callback=hf_writer,
    description=description,
    article=article,
    examples=examples,
    theme="grass",
)

iface.launch()
```

**Comment by kingabzpro** [comment]
@abidlabs, it worked locally. But my return is Urdu text. If you replace "hello world" with "آپ کیا کر ر" then you will see the error. I think it is also causing HFcallback error. Unicode decode. I tried to normalize text but still face and error.


`UnicodeEncodeError: 'charmap' codec can't encode characters in position 134-135: character maps to <undefined>`

**Comment by abidlabs** [comment]
Ahh I see

**Comment by kingabzpro** [comment]
@abidlabs I was able to normalize text, but apart from that, there is another but strange error. When you selected examples. 
![image](https://user-images.githubusercontent.com/36753484/159248049-7083efae-87fb-4bcc-b37d-2ac43db3a710.png)

```bash
    filepath = component.save_flagged(
  File "/home/user/.local/lib/python3.8/site-packages/gradio/inputs.py", line 1295, in save_flagged
    return self.save_flagged_file(
  File "/home/user/.local/lib/python3.8/site-packages/gradio/component.py", line 60, in save_flagged_file
    file = processing_utils.decode_base64_to_file(data, encryption_key)
  File "/home/user/.local/lib/python3.8/site-packages/gradio/processing_utils.py", line 187, in decode_base64_to_file
    data, extension = decode_base64_to_binary(encoding)
  File "/home/user/.local/lib/python3.8/site-packages/gradio/processing_utils.py", line 182, in decode_base64_to_binary
    data = encoding.split(",")[1]
```

This error occurs when you select the examples and flag them.

**Comment by FarukOzderim** [comment]
@abidlabs Assigning it to you since you are taking a look into it, if not feel free to remove it.

**Comment by abidlabs** [comment]
Hi @kingabzpro this fixed now, and the fix will be out as part of Gradio 3.0. Thanks for creating the issue!
